### PR TITLE
Fix some code quality and bug-risk issues

### PR DESCRIPTION
**Pull request by pnijhara**
- Fix dangerous default argument
- Remove unnecessary comprehension
---

Find the other issues found here - [https://deepsource.io/gh/pnijhara/xarray/issues/?category=all](https://deepsource.io/gh/pnijhara/xarray/issues/?category=all)
This PR also adds `.deepsource.toml` configuration file to run DeepSource analysis on the repo with. Upon enabling DeepSource, the analysis will run on every PR and commit to detect 560+ types of issues in the changes — including bug risks, anti-patterns, security vulnerabilities, etc.

To enable DeepSource analysis after merging this PR, please follow these steps:

1. [Signup](https://deepsource.io/signup/) on DeepSource with your GitHub account and grant access to this repo.
2. Activate analysis on this repo [here](http://deepsource.io/gh/pydata/xarray)

You can also look at the [docs](https://deepsource.io/docs/guides/quickstart.html) for more details. Do let me know if I can be of any help!

### CODE REVIEW

**Comment by pnijhara** [comment]
@shoyer Thanks for the suggestions. I have re-introduced len() over there. Seems like tests were written by taking dates as numpy array and not just simple array.

File: .deepsource.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    "doc/**",
+    "ci/**"
+]
+
+[[analyzers]]
+name = "python"
+enabled = true
+
+  [analyzers.meta]
+  runtime_version = "3.x.x"
```
Comment (shoyer) [inline .deepsource.toml:18]: Is this necessary for deepsource to give useful results? I guess we can give it a try if you want to keep on making contributions using it, but I don't want to add it to CI at this time and if it doesn't get much use we'll probably eventually remove it.

File: MANIFEST.in [hunk 2]
```diff
@@ -1,4 +1,5 @@
 include LICENSE
+include *.toml
```
Comment (shoyer) [inline MANIFEST.in:2]: I don't think we want to include this file in source distributions on pypi, so please remove this line.

File: .deepsource.toml (hunk 1 above)
Comment (pnijhara) [reply .deepsource.toml:18]: Sure

**Comment by shoyer** [comment]
Thanks

**Pull request merged**